File: static/code/sample/tethered/gdl-ldleds564-a.py
# In this sample:
# Display the current minute and second on the LED (timer version).

import time
from datetime import datetime
from DUELink.DUELinkController import DUELinkController

availablePort = DUELinkController.GetConnectionPort()
duelink = DUELinkController(availablePort)

# method
def DrawNumber(led, number):
    duelink.Engine.ExecuteCommand(f"Seg7({led},{number})")

def ShowColon(value):
    duelink.Engine.ExecuteCommand(f"Colon({value})")

def Show():
    duelink.Engine.ExecuteCommand("Show()")

min = 0
sec = 0

while True:
    now = datetime.utcnow()
    minute = now.minute
    second = now.second

    if sec != second or min != minute:
        sec = second
        min = minute
        DrawNumber(0, sec % 10)
        DrawNumber(1, sec // 10)

        DrawNumber(2, min % 10)
        DrawNumber(3, min // 10)

    ShowColon(1)
    Show()
    time.sleep(0.4)  # 

    ShowColon(0)
    Show()
    time.sleep(0.4)  # 

File: static/code/sample/tethered/gdl-ldleds564-a.js
// In this sample:
// Display the current minute and second on the LED (timer version).

import pkg_serialusb from 'dlserialusb';
const {SerialUSB} = pkg_serialusb

import pkg_duelink from 'duelink';
const {DUELinkController} = pkg_duelink

let duelink = new DUELinkController(new SerialUSB());
await duelink.Connect();

// Sleep helper (ms)
function sleep(ms) {
    return new Promise(resolve => setTimeout(resolve, ms));
}

// Methods
async function DrawNumber(led, number) {
    await duelink.Engine.ExecuteCommand(`Seg7(${led},${number})`);
}

async function ShowColon(value) {
    await duelink.Engine.ExecuteCommand(`Colon(${value})`);
}

async function Show() {
    await duelink.Engine.ExecuteCommand("Show()");
}

let min = 0;
let sec = 0;

(async function mainLoop() {
    while (true) {
        const now = new Date();
        const minute = now.getUTCMinutes();
        const second = now.getUTCSeconds();

        if (sec !== second || min !== minute) {
            sec = second;
            min = minute;

            await DrawNumber(0, sec % 10);
            await DrawNumber(1, Math.floor(sec / 10));

            await DrawNumber(2, min % 10);
            await DrawNumber(3, Math.floor(min / 10));
        }

        await ShowColon(1);
        await Show();
        await sleep(400); // 

        await ShowColon(0);
        await Show();
        await sleep(400); // 
    }
})();
